lib/ecto/schema.ex

```diff
@@ -1139,9 +1196,66 @@ defmodule Ecto.Schema do
       # Update the order
       changeset = Repo.update!(changeset)
 
+  ## Inline embedded schema
+
+  The schema module can be defined inline in the parent schema in simple
+  cases:
+
+      defmodule Parent do
+        use Ecto.Schema
+
+        schema "parents" do
+          field :name, :string
+
+          embeds_one :children, Child do
+            field :name, :string
+            field :age,  :integer
+          end
+        end
+      end
+
+  Defining embedded schema in such a way will define a `Parent.Child` module
+  with the appropriate struct. In order to properly cast the embedded schema.
+  When casting the inline-defined embedded schemas you need to use the `:with`
+  option of `cast_embed/3` to provide the proper function to do the casting.
+  For example:
+
+      def changeset(schema, params) do
+        schema
+        |> cast(params, [:name])
+        |> cast_embed(:children, with: &child_changeset/2)
```

Comment:
The current error will be along the lines of:

```
     ** (UndefinedFunctionError) function Module.changeset/2 is undefined or private. Did you mean one of:

           * __changeset__/0
```

which, I agree, is not the friendliest. 

If we were to raise something we should consider doing something in [here](https://github.com/elixir-ecto/ecto/blob/042aa92c62883572917477b2acaad3297590775c/lib/ecto/changeset.ex#L557), where we assume a `changeset/2` function will exist.